source/controllers/src/impedance/Impedance.cpp

```diff
@@ -32,29 +32,35 @@ template<>
 CartesianState Impedance<CartesianState>::compute_command(const CartesianState& desired_state,
                                                           const CartesianState& feedback_state) {
   CartesianState state_error = desired_state - feedback_state;
-  // compute the wrench using the formula W = K * e_pos + D * e_vel + I * acc_desired
+  // compute the wrench using the formula W = I * acc_desired + K * e_pose + D * e_twist - e_wrench
   CartesianWrench command(feedback_state.get_name(), feedback_state.get_reference_frame());
   // compute force
-  command.set_force(this->get_stiffness().block<3, 3>(0, 0) * state_error.get_position()
-                        + this->get_damping().block<3, 3>(0, 0) * state_error.get_linear_velocity()
-                        + this->get_inertia().block<3, 3>(0, 0) * desired_state.get_linear_acceleration());
+  Eigen::Vector3d force_control = this->get_stiffness().block<3, 3>(0, 0) * state_error.get_position()
```

Comment:
I'd prefer using `topLeftCorner<3,3>()` and `bottomRightCorner<3,3>()` instead of the `block`s here because I think it makes the code more readable but this is just a personal preference

Reply:
not only that but it is also far less error prone. Very good point